include/device_vector.h

```diff
@@ -0,0 +1,150 @@
+#pragma once
+
+#include <vector>
+
+#include <quda_api.h>
+#include <malloc_quda.h>
+#include <util_quda.h>
+#include <complex_quda.h>
+
+/**
+   @file device_vector.h
+   Helper functionality for handling a small piece of device memory.
+ */
+
+namespace quda
+{
+  /**
+    @brief This class `device_vector` provides a light weighted wrapper for data on device.
+  */
+
+  template <class real> class device_vector
```

Comment:
I'm not going to object to this addition, as it seems very useful, but the addition of an object named `device_vector` seems antithetical to the current drive to unify objects, e.g., `cudaColorspinorField` / `cpuColorspinorField`.  I reserve the right to revisit this in the future 😜